Seomeone picked half of 215bebc8c6ac438c382a6a56bd2764a2d4e1da72 but didn't want the other half...
---
--- Visionfive2-kernel-5.15.0/security/keys/dh.c	2023-01-11 16:29:35.000000000 +0100
+++ -	2023-02-01 22:15:10.262387290 +0100
@@ -14,7 +14,7 @@
 #include <keys/user-type.h>
 #include "internal.h"
 
-static ssize_t dh_data_from_key(key_serial_t keyid, void **data)
+static ssize_t dh_data_from_key(key_serial_t keyid, const void **data)
 {
 	struct key *key;
 	key_ref_t key_ref;
